pkg/storage/storage_test.go

```diff
@@ -0,0 +1,327 @@
+package storage
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"testing"
+)
+
+func TestStorage_AddProfile(t *testing.T) {
+	tests := []struct {
+		name       string
+		storage    *Storage
+		newProfile *Profile
+		validate   func(*testing.T, *Storage)
+	}{
+		{
+			name: "add new Profile",
+			storage: &Storage{
+				Default: "default",
+				Profiles: []*Profile{
+					{Name: "default", Key: "key"},
+				},
+			},
+			newProfile: &Profile{Name: "test", Key: "key"},
+			validate: func(t *testing.T, s *Storage) {
+				if len(s.Profiles) != 2 {
+					t.Fatalf("got %d profiles, want 2", len(s.Profiles))
+				}
+				if s.Profiles[1].Name != "test" {
+					t.Fatalf("got %s, want %s", s.Profiles[1].Name, "test")
+				}
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tt.storage.AddProfile(tt.newProfile)
+			tt.validate(t, tt.storage)
+		})
+	}
+}
+
+func TestStorageExtractProfile(t *testing.T) {
+
+	tests := []struct {
+		name     string
+		storage  *Storage
+		wantName string
+		wantErr  bool
+		validate func(*testing.T, *Profile)
+	}{
+		{
+			name: "existing profile",
+			storage: &Storage{
+				Default: "default",
+				Profiles: []*Profile{
+					{Name: "default", Key: "key"},
+					{Name: "test", Key: "key"},
+				},
+			},
+			wantName: "test",
+			wantErr:  false,
+			validate: func(t *testing.T, p *Profile) {
+				if p == nil {
+					t.Fatalf("got nil, want %s", "test")
+				}
+				if p.Name != "test" {
+					t.Fatalf("got %s, want %s", p.Name, "test")
+				}
+			},
+		},
+		{
+			name: "not found profile",
+			storage: &Storage{
+				Default: "default",
+				Profiles: []*Profile{
+					{Name: "default", Key: "key"},
+					{Name: "test", Key: "key"},
+				},
+			},
+			wantName: "test2",
+			wantErr:  true,
+			validate: func(t *testing.T, p *Profile) {},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := tt.storage.ExtractProfile(tt.wantName)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("Storage.ExtractProfile() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			tt.validate(t, got)
+		})
+	}
+}
+
+func TestStorage_Save(t *testing.T) {
+	// Create a temporary directory for tests
+	tempDir := t.TempDir()
+
+	tests := []struct {
+		name     string
+		storage  *Storage
+		wantErr  bool
+		validate func(*testing.T, string)
+	}{
+		{
+			name: "successful save with default profile",
+			storage: &Storage{
+				Default:     "test_profile",
+				Profiles:    []*Profile{{Name: "test_profile", Key: "test_key"}},
+				StoragePath: fmt.Sprintf("%s/config1.json", tempDir),
```

Comment:
path.Join ?

And one more thing: if you're not going to write files, you are safe to use test data (often implemented as reading files from testdata directory right at test files)